tests/integration_tests/conftest.py

```diff
@@ -40,29 +39,51 @@
 UNFCCC_ORG_ID = 2
 SUPER_ORG_ID = 50
 
+migration_file = None
 
-def get_test_db_url() -> str:
-    return SQLALCHEMY_DATABASE_URI + f"_test_{uuid.uuid4()}"
 
+def _create_engine_run_migrations(test_db_url: str):
+    test_engine = create_engine(test_db_url)
+    run_migrations(test_engine)
+    return test_engine
 
-@pytest.fixture(scope="function")
-def slow_db(monkeypatch):
-    """Create a fresh test database for each test."""
 
-    test_db_url = get_test_db_url()
+def do_cached_migrations(test_db_url: str):
+
+    global migration_file  # Note this is scoped to the module, so it will not get recreated.
 
     # Create the test database
     if database_exists(test_db_url):
         drop_database(test_db_url)
     create_database(test_db_url)
 
+    test_engine = None
+
+    if not migration_file:
+        test_engine = _create_engine_run_migrations(test_db_url)
+        migration_file = tempfile.NamedTemporaryFile().name
+        result = subprocess.run(["pg_dump", "-f", migration_file])
+        assert result.returncode == 0
+    else:
+        result = subprocess.run(["psql", "-f", migration_file])
+        assert result.returncode == 0
+        test_engine = create_engine(test_db_url)
+
+    return test_engine
+
+
+@pytest.fixture(scope="function")
+def slow_db(monkeypatch):
```

Comment:
A bit pedantic, but shouldn't this now be a quick_db rather than a slow DB if the migrations are cached?

Reply:
Ah fair point ... I can probably remove that  now tbf :eye: